public/js/reports/reports.service.js

```diff
@@ -79,5 +80,33 @@
 
             return columnDescription;
         };
+
+        function checkPrompts (prompts) {
+            const promptsStatus = {};
```

Comment:
Only the values of `promptsStatus` seem to be useful. Why use an object instead of an array ?

Reply:
In fact, I think you don't need `promptsStatus` variable at all. You can just wrap the whole thing inside a call to `Array.prototype.every`, like this:

```js
return prompts.every(p => {
    let isValid;
    // ...
    return isValid;
});
```

and to make things even more clear, you can move the checks for a single prompt inside its own function, so that would become:

```js
function checkPrompts (prompts) {
    return prompts.every(checkPrompt)
}
```

This also has the benefit of "exiting the loop" as soon as one prompt is found to be invalid (the other prompts don't need to be checked in that case)

Reply:
that's clearer, I fix it